gallery/gallery/lib/demos/material/progress_indicator_demo.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN progressIndicatorsDemo
+
+enum ProgressIndicatorDemoType {
+  circular,
+  linear,
+}
+
+class ProgressIndicatorDemo extends StatefulWidget {
+  const ProgressIndicatorDemo({Key key, this.type}) : super(key: key);
+
+  final ProgressIndicatorDemoType type;
+
+  @override
+  _ProgressIndicatorDemoState createState() => _ProgressIndicatorDemoState();
+}
+
+class _ProgressIndicatorDemoState extends State<ProgressIndicatorDemo>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation<double> _animation;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 1500),
+      vsync: this,
+      animationBehavior: AnimationBehavior.preserve,
+    )..forward();
+
+    _animation = CurvedAnimation(
+      parent: _controller,
+      curve: const Interval(0.0, 0.9, curve: Curves.fastOutSlowIn),
+      reverseCurve: Curves.fastOutSlowIn,
+    )..addStatusListener((status) {
+        if (status == AnimationStatus.dismissed) {
+          _controller.forward();
+        } else if (status == AnimationStatus.completed) {
+          _controller.reverse();
+        }
+      });
+  }
+
+  @override
+  void dispose() {
+    _controller.stop();
+    super.dispose();
+  }
+
+  String _title(BuildContext context) {
+    switch (widget.type) {
+      case ProgressIndicatorDemoType.circular:
+        return GalleryLocalizations.of(context)
+            .demoCircularProgressIndicatorTitle;
+      case ProgressIndicatorDemoType.linear:
+        return GalleryLocalizations.of(context)
+            .demoLinearProgressIndicatorTitle;
+    }
+    return '';
+  }
+
+  void _handleTap() {
+    setState(() {
+      // valueAnimation.isAnimating is part of our build state
+      if (_controller.isAnimating) {
+        _controller.stop();
+      } else {
+        switch (_controller.status) {
```

Comment:
I feel like this logic is done twice (here and in `initState`), maybe you could make a function like `startOrStopAnimation`?

Reply:
This was slightly different in order to handle stop/starting the animation when you tap. But this logic didn't add much value to the demo so I just removed it instead